src/utils.js

```diff
@@ -844,8 +844,24 @@ export function getCookie(name) {
   return m ? decodeURIComponent(m[2]) : null;
 }
 
-export function setCookie(key, value, expires, sameSite) {
-  document.cookie = `${key}=${encodeURIComponent(value)}${(expires !== '') ? `; expires=${expires}` : ''}; path=/${sameSite ? `; SameSite=${sameSite}` : ''}`;
+export function setCookie(key, value, expires, sameSite, domain) {
+  const domainPortion = (domain && domain !== '') ? ` ;domain=${encodeURIComponent(domain)}` : ''
+  document.cookie = `${key}=${encodeURIComponent(value)}${(expires !== '') ? `; expires=${expires}` : ''}; path=/${domainPortion}${sameSite ? `; SameSite=${sameSite}` : ''}`;
+}
+
+export function findSimilarCookies(keyLike) {
+  const all = [];
+  const cookies = document.cookie.split(';');
+  while (cookies.length) {
+    const cookie = cookies.pop();
+    let separatorIndex = cookie.indexOf('=');
+    separatorIndex = separatorIndex < 0 ? cookie.length : separatorIndex;
+    const cookieName = decodeURIComponent(cookie.slice(0, separatorIndex).replace(/^\s+/, ''));
+    if (cookieName.indexOf(keyLike) >= 0) {
+      all.push(decodeURIComponent(cookie.slice(separatorIndex + 1)));
+    }
+  }
+  return all;
```

Comment:
Hi @jankoulaga 

I noticed with the recent set of commits, this new function was added to the `utils` file.  I saw there was a unit test that used this function, but it seems that the test was later removed.  

Should this function be removed as well?  Or will there be some future update were it will be used?

Reply:
Hey @jsnellbaker true... I added this spec in order to provide coverage, however, the spec did not pass in browserstack for a reason unknown to me.

When i execute this exact snippet in IE 11 && Win 10, it passes without a glitch, so it was hard for me to understand what the actual problem was.

As the utils object is now used in live-connect-js, this function is necessary. 

Let's make a deal :) If you're in general ok with the approach we're taking here, i'll gladly work on re-adding a spec for this function ASAP. How does that sound?

Reply:
I think the approach used for this function is fine.  

If you can add the test back in, we can try to help with the debugging if there's something strange going with the BrowserStack tests.  

I might also suggest adding a comment before the function to note that it is used by the live-connect package, so it's not altered/removed accidentally later (ie if someone tried to clean-up the utils functions by removing ones not used in any of the project `src` or `modules` files).

Reply:
Cool, @jsnellbaker ... I will re-add the specs and figure out what's wrong, add all the right comments and i'll update the PR soon. In the mean time, we'd appreciate more input on what can/has to be fixed so we can speed things up and make the review process faster, so we don't have to address individual comments in multiple commits, if that's ok with you.

Reply:
I looked at the rest of the changes and added notes, and I think those are fine.

The only other part I want to confirm is related to the consent information.  I saw you're reading the USP values to pass along the consent string.  Would you want to do this for the GDPR/TCF info as well?  Or is that definitely not in-scope for your use cases?

Reply:
I don't think it applies for now, as in general, we're fine with the approach of the module not being initialized if there's no GDPR consent given. It's food for thought, but as this was not implemented initially we didn't touch it. The scope of this PR was to introduce a more valuable feature, and i think we can re-iterate on the GDPR topic in the future.

Reply:
Ok, thanks for the confirmation.

As a follow-up to the IE11 test issue, I did a little digging and think I found the issue.  IE11 doesn't like cookies that have a blank (or undefined) value for the `expires` field.  In that scenario, it was advised to not have the `expires` field-name at all when writing the cookie.

I did a local test where I updated the test's setCookie functions to pass `''` in the expires argument and the tests passed in BrowserStack.

Can you give this a try?

Reply:
@jsnellbaker i was just about to push that :) I guess the cookie part of the utils module needs a bit more tests ;). Let's see how it goes.

Reply:
@jsnellbaker i hope you can trigger cci re-builds, as the current one seems to be stale on BS

Reply:
I have restarted the job and the IE11 failure did not occur.  The only failure that did happen was unreleated to the changes made by this PR, so that's fine.